sphinx/ext/intersphinx.py

```diff
@@ -649,36 +680,76 @@ def install_dispatcher(app: Sphinx, docname: str, source: list[str]) -> None:
 
 
 def normalize_intersphinx_mapping(app: Sphinx, config: Config) -> None:
-    for key, value in config.intersphinx_mapping.copy().items():
+    # URIs should NOT be duplicated, otherwise different builds may use
+    # different project names (and thus, the build are no more reproducible)
+    # depending on which one is inserted last in the cache.
+    seen: dict[InventoryURI, InventoryName] = {}
+
+    for name, value in config.intersphinx_mapping.copy().items():
+        if not isinstance(name, str):
+            logger.warning(
+                __('intersphinx identifier %r is not string. Ignored'),
+                name, type='intersphinx', subtype='config',
+            )
+            del config.intersphinx_mapping[name]
+            continue
+
+        # ensure that intersphinx projects are always named
+        if not name:
+            logger.warning(
+                __('ignoring empty intersphinx identifier'),
+                type='intersphinx', subtype='config',
+            )
+            del config.intersphinx_mapping[name]
+            continue
+
+        if not isinstance(value, (tuple, list)):
+            logger.warning(
+                __('intersphinx_mapping[%r]: expecting a tuple or a list, got: %r; ignoring.'),
+                name, value, type='intersphinx', subtype='config',
+            )
+            del config.intersphinx_mapping[name]
+            continue
+
         try:
-            if isinstance(value, (list, tuple)):
-                # new format
-                name, (uri, inv) = key, value
-                if not isinstance(name, str):
-                    logger.warning(__('intersphinx identifier %r is not string. Ignored'),
-                                   name)
-                    config.intersphinx_mapping.pop(key)
-                    continue
+            uri, inv = value
```

Comment:
```suggestion
            uri, inv = value
            inv = inv if isinstance(inv, (tuple, list)) else (inv,)
```

(to be combined with simplification of L742)

Reply:
I want to isolate the unpack statement from the assignment itself and I want to first check the type of *uri* before doing anything on the *inv* part.